packages/core/src/middleware/koa-role-guard.ts

```diff
@@ -0,0 +1,28 @@
+import { UserRole } from '@logto/schemas';
+import { MiddlewareType } from 'koa';
+
+import RequestError from '@/errors/RequestError';
+import { findUserById } from '@/queries/user';
+import assertThat from '@/utils/assert-that';
+
+import { WithAuthContext } from './koa-auth';
+
+export default function koaRoleGuard<
+  StateT,
+  ContextT extends WithAuthContext,
+  ResponseBodyT
+>(): MiddlewareType<StateT, ContextT, ResponseBodyT> {
+  return async (ctx, next) => {
+    const userId = ctx.auth;
+    const userInfo = await findUserById(userId).catch(() => {
+      throw new RequestError({ code: 'auth.unauthorized', status: 401 });
+    });
```

Comment:
should this be already set in `koaUserInfo()`? or we can remove `koaUserInfo()` and keep this. both are good to me

Reply:
nice catch removed here. should leverage the `koaUserInfo()` middleware